backend/src/cusco/storage.py

```
@@ -0,0 +1,312 @@
+            except Exception:
+                conn.execute("ROLLBACK")
+                raise
+        logger.info(f"SQLite table '{self.name}': replaced with {count} rows")
+        return count
+
+    def get_by_nif(self, nif: str) -> list[dict]:
+        """Look up all payloads for a given NIF. Returns [] on any error."""
+        try:
+            with get_connection() as conn:
+                cur = conn.execute(
+                    f'SELECT payload FROM "{self.name}" WHERE nif = ?', (nif,)
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)